xarray/backends/zarr.py

```diff
@@ -666,6 +671,8 @@ def set_variables(self, variables, check_encoding_set, writer, unlimited_dims=No
                 # metadata. This would need some case work properly with region
                 # and append_dim.
                 zarr_array = self.zarr_group[name]
+                if self._write_empty is not None:
+                    zarr_array._write_empty_chunks = self._write_empty
```

Comment:
Looks like a non-public attribute? There's no public alternative?

Reply:
Not that I could find. There's a way to specify it when opening direct arrays but since xarray goes through `zarr.hierarchy.open_group` (and there is no equivalent parameter there compared to `zarr.creation.open_array`, nor kwargs to pass) there is no way to do so, at least from within xarray.

Reply:
This makes me nervous. How important is it that we support this feature on existing arrays? Alternatively, we can work on making the `write_empty_chunks` property in Zarr public.

Reply:
Another option is to change how you create the array. Instead of using `zarr_array = self.zarr_group[name]`, you can create the array directly, e.g. 
```python
zarr_array = zarr.open(store=self.zarr_group.store, path=f'{self.zarr_group.name}/{name}', write_empty_chunks=...)
```
This is longer, but you have complete control over the instantiation of the zarr array

Reply:
I'll give that a try, may be a good solution pending zarr-developers/zarr-python#1478